pkg/kafka/auth/config.go

```diff
@@ -89,3 +90,14 @@ func (config *AuthenticationConfig) InitFromViper(configPrefix string, v *viper.
 	config.PlainText.UserName = v.GetString(configPrefix + plainTextPrefix + suffixPlainTextUserName)
 	config.PlainText.Password = v.GetString(configPrefix + plainTextPrefix + suffixPlainTextPassword)
 }
+
+// Normalize normalizes kafka options
+func (config *AuthenticationConfig) Normalize(logger *zap.Logger) {
```

Comment:
I had to add to split `InitFromViper` and `Normalize`. `InitFromViper` is called via storage factory and it does not have access to the logger when doing the call.